invoices/invoice_expiry_watcher.go

```diff
@@ -119,20 +118,54 @@ func (ew *InvoiceExpiryWatcher) AddInvoice(
 	}
 
 	expiry := invoice.CreationDate.Add(realExpiry)
-
-	log.Debugf("Adding invoice '%v' to expiry watcher, expiration: %v",
-		paymentHash, expiry)
-
-	newInvoiceExpiry := &invoiceExpiry{
+	return &invoiceExpiry{
 		PaymentHash: paymentHash,
 		Expiry:      expiry,
 	}
+}
+
+// AddInvoices adds multiple invoices to the InvoiceExpiryWatcher.
+func (ew *InvoiceExpiryWatcher) AddInvoices(
+	invoices []channeldb.InvoiceWithPaymentHash) {
+
+	var invoicesWithExpiry []*invoiceExpiry
```

Comment:
We can pre-size this slice as we know ahead of time how many invoices we'll need to be sending over.

Reply:
Invoices not in ContractOpen state won't be added but at least we know the capacity. Fixed.